src/main/java/org/testmonkeys/jentitytest/comparison/entity/EntityComparator.java

```diff
@@ -28,7 +28,9 @@
         List<ComparisonResult> comparisonResults = new LinkedList<>();
         if (context == null) {
             context = new ComparisonContext();
-            context.setParentName("Entity");
+            String parent = expected != null ? expected.getClass().getSimpleName() :
```

Comment:
too complicated logical statement. Extract to a method, and split logical statement.